### PR TITLE
feature / invitation

### PR DESCRIPTION
**Pull request by pvilalta**
https://platform-dev-invitation.bimdata.io

### CODE REVIEW

**Review by Kurtil — APPROVED** [review]

File: src/config/invitation.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const INVITATION_VALIDATION_STATUS = Object.freeze({
+  PENDING: "P",
+  ACCEPT: "A",
```
Comment (Amoki) [inline src/config/invitation.js]: ACCEPTED

File: src/config/invitation.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  DENY: "D"
```
Comment (Amoki) [inline src/config/invitation.js]: DENIED

File: src/views/invitation/Invitation.vue [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    const acceptAllInvitation = async () => {
+      await async.eachLimit(
+        invitationListPending.value,
+        20,
+        onAcceptInvitation
+      );
+      await fetchInvitations();
+    };
+
+    onMounted(async () => fetchInvitations());
```
Comment (Amoki) [inline src/views/invitation/Invitation.vue]: useless async

File: src/views/invitation/Invitation.vue [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+              >
+                <div class="invitation__content__list__invit__button__pending">
+                  <BIMDataButton
+                    class="invitation__content__list__invit__button__pending--deny"
+                    width="35px"
+                    height="35px"
+                    radius
+                    ghost
+                    icon
+                    @click="onDenyInvitation(invit).then(fetchInvitations)"
```
Comment (Amoki) [inline src/views/invitation/Invitation.vue]: Handling async stuff in the template seems complex.
Do we want to call `fetchInvitations()` in the function `onDenyInvitation()` ?
@Kurtil

File: src/config/invitation.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const INVITATION_VALIDATION_STATUS = Object.freeze({
```
Comment (Amoki) [inline src/config/invitation.js]: `INVITATION_STATUS`
There is no "validation" concept in an invitation imho

**Review by Amoki — CHANGES_REQUESTED** [review]

**Comment by pvilalta** [comment]
PR in draft mod in order to resolve all team feedbacks

File: src/router/guards/data.js [hunk 6]
```diff
@@ -26,8 +28,12 @@ export default async function dataGuard() {
...
     await loadUserOrganizations();
 
-    // 3) Load spaces and projects of the current user
-    await Promise.all([loadUserSpaces(), loadUserProjects()]);
+    // 3) Load spaces, projects and invitations of the current user
+    await Promise.all([
+      loadUserSpaces(),
+      loadUserProjects(),
+      loadUserInvitations()
+    ]);
```
Comment (Kurtil) [inline src/router/guards/data.js:36]: I don't think the comments are usefull here... The functions are well named, and you await them so yes it is important to await them before calling the following. The comments are juste describing what the code already tells you :)

**Review by Kurtil — CHANGES_REQUESTED** [review]

**Review by Kurtil — APPROVED** [review]

File: src/router/guards/data.js (hunk 6 above)
Comment (Kurtil) [reply src/router/guards/data.js:36]: The code I commented is not from this PR... my bad!

**Review by Amoki — APPROVED** [review]

File: src/router/index.js [hunk 7]
```diff
@@ -93,6 +96,14 @@ const routes = [
           back: dashboard
         }
       },
+      {
+        path: "/invitation",
```
Comment (NicolasRichel) [inline src/router/index.js]: To be consistent with the general naming convention of the project I would recommend to use a plural form for this route: `/invitations`.

File: src/router/index.js [hunk 8]
```diff
@@ -93,6 +96,14 @@ const routes = [
...
+        name: invitation,
+        component: Invitation,
```
Comment (NicolasRichel) [inline src/router/index.js]: The same apply to route name and component name: `Invitations`.

File: src/state/invitations.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+const acceptInvitations = async invitations => {
+  await async.eachLimit(
+    invitations,
+    20,
+    InvitationViewService.acceptInvitation
+  );
+  await loadUser();
+  await Promise.all([loadUserSpaces(), loadUserProjects()]);
+  await loadUserInvitations();
```
Comment (NicolasRichel) [inline src/state/invitations.js:37]: Why does `loadUserInvitations()` isn't inside `Promise.all` ?
Is it necessary to call it seperately ?

File: src/views/invitation/Invitation.vue [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+      INVITATION_STATUS,
+      isLoadingAllInvit,
+      currentInvitation,
+      invitationListPending,
+      isLoadingSingleInvit,
+      // methods
+      fullName,
+      onDenyInvitation,
+      onAcceptInvitations,
+      getBack: () => router.back(),
```
Comment (NicolasRichel) [inline src/views/invitation/Invitation.vue]: Is this method still needed ? I can't find were you use it.

**Review by NicolasRichel — CHANGES_REQUESTED** [review]

**Review by NicolasRichel — APPROVED** [review]

**Review by GabBimdata — APPROVED** [review]

**Pull request merged**